firebase.go

```diff
@@ -99,6 +99,11 @@ func New(url string) *Firebase {
 		CheckRedirect: redirectPreserveHeaders,
 	}
 
+	return NewFromHttpClient(url, client)
+}
+
+// NewFromHttpClient creates a new Firebase reference with given http client
+func NewFromHttpClient(url string, client *http.Client) *Firebase {
```

Comment:
If we expose Client we need a way to make concurrent changes safe (aka mutex)
I'd go with what @hostirosti proposes, it's far less dangerous :smile: